src/LakatArticleMetadata.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace MediaWiki\Extension\Lakat;
+
+use CommentStoreComment;
+use ContentHandler;
+use Exception;
+use FormatJson;
+use JsonContent;
+use MediaWiki\Language\RawMessage;
+use MediaWiki\MediaWikiServices;
+use MediaWiki\Permissions\Authority;
+use MediaWiki\Revision\RevisionRecord;
+use MediaWiki\Title\Title;
+use MediaWiki\User\UserIdentity;
+use WikiPage;
+
+class LakatArticleMetadata {
+	public static function getBranchId( string $branchName ) {
+		$page = MediaWikiServices::getInstance()->getWikiPageFactory()->newFromTitle( Title::newFromText($branchName) );
+		if ( !$page->exists() ) {
+			throw new Exception("Branch page doesn't exist");
+		}
+		$metadata = self::getPageMetadata($page);
+		if (!isset($metadata->BranchId)) {
+			throw new Exception('Invalid metadata: BranchId field is not set');
+		}
+		return $metadata->BranchId;
+	}
+
+	public static function saveArticleId( WikiPage $wikiPage, Authority|UserIdentity $user, string $articleId): void
+	{
+		$articleMetadataContent = ContentHandler::makeContent( FormatJson::encode(compact('articleId')), null, CONTENT_MODEL_JSON);
+		$pageUpdater = $wikiPage->newPageUpdater( $user );
+		$pageUpdater->setContent('lakat', $articleMetadataContent);
+		$pageUpdater->saveRevision(CommentStoreComment::newUnsavedComment('Lakat: added article metadata'), EDIT_SUPPRESS_RC);
+	}
+
+	public static function getArticleId( WikiPage $wikiPage ): string {
+		$metadata = self::getPageMetadata($wikiPage);
+		if (!isset($metadata->articleId)) {
+			throw new Exception('Article has invalid metadata: articleId field is not set');
+		}
+		return $metadata->articleId;
+	}
+
+	private static function getPageMetadata(WikiPage $page): object {
+		$revisionRecord = $page->getRevisionRecord();
+		if (!$revisionRecord->hasSlot('lakat')) {
+			throw new Exception('Page has no metadata slot');
+		}
```

Comment:
So the logic here is that the slot lakat is created for each now branch?

Reply:
Yes, for each branch and for each article in a branch.